contracts/EscrowMigrator.sol

```diff
@@ -0,0 +1,453 @@
+        external
+        view
+        override
+        returns (uint256[] memory)
+    {
+        uint256 endIndex = _index + _pageSize;
+
+        // If the page extends past the end of the list, truncate it.
+        uint256 numEntries = numberOfRegisteredEntries(_account);
+        if (endIndex > numEntries) {
+            endIndex = numEntries;
+        }
+        if (endIndex <= _index) {
+            return new uint256[](0);
+        }
+
+        uint256 n = endIndex - _index;
+        uint256[] memory page = new uint256[](n);
+        for (uint256 i; i < n;) {
+            unchecked {
+                page[i] = registeredEntryIDs[_account][i + _index];
+            }
+
+            unchecked {
+                ++i;
+            }
+        }
+        return page;
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                                 STEP 0
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice claim any remaining StakingRewards V1 rewards
+    /// This must be done before the migration process can begin
+
+    /*//////////////////////////////////////////////////////////////
+                                 STEP 1
+    //////////////////////////////////////////////////////////////*/
+
+    /// @inheritdoc IEscrowMigrator
+    function registerEntries(uint256[] calldata _entryIDs) external override {
+        _registerEntries(msg.sender, _entryIDs);
+    }
+
+    function _registerEntries(address _account, uint256[] calldata _entryIDs)
+        internal
+        whenNotPaused
+    {
+        if (!initiated[_account]) {
+            if (stakingRewardsV1.earned(_account) != 0) revert MustClaimStakingRewards();
+            if (rewardEscrowV1.balanceOf(_account) == 0) revert NoEscrowBalanceToMigrate();
+
+            initiated[_account] = true;
+            escrowVestedAtStart[_account] = rewardEscrowV1.totalVestedAccountBalance(_account);
+        }
+
+        uint256 registeredEscrow;
+        for (uint256 i = 0; i < _entryIDs.length; i++) {
+            uint256 entryID = _entryIDs[i];
+
+            // skip if already registered
+            if (registeredVestingSchedules[_account][entryID].escrowAmount != 0) continue;
+
+            (, uint256 escrowAmount,) =
+                rewardEscrowV1.getVestingEntry(_account, entryID);
+
+            // skip if entry is already vested or does not exist
+            if (escrowAmount == 0) continue;
+
+            registeredVestingSchedules[_account][entryID] =
+                VestingEntry({escrowAmount: escrowAmount, migrated: false});
+
+            /// @dev A counter of numberOfRegisteredEntries would do, but this allows easier inspection
+            registeredEntryIDs[_account].push(entryID);
+            registeredEscrow += escrowAmount;
+        }
+
+        /// @dev Simlarly this value is not needed, but just added for easier on-chain inspection
+        totalRegistered += registeredEscrow;
```

Comment:
Hmm events + subgraph are usually useful for things like this. Is this used for tests?

Reply:
@JChiaramonte7 This will be needed in the `recoverFunds` function, so I guess I need to remove the comment once that function is added.